operator/src/main/java/oracle/kubernetes/operator/EventConstants.java

```diff
@@ -6,35 +6,34 @@
 /** Constants used in Operator events. */
 public interface EventConstants {
 
-  String DOMAIN_CHANGED_EVENT = "DomainChanged";
-  String DOMAIN_CREATED_EVENT = "DomainCreated";
-  String DOMAIN_DELETED_EVENT = "DomainDeleted";
-  String DOMAIN_PROCESSING_COMPLETED_EVENT = "DomainProcessingCompleted";
-  String DOMAIN_AVAILABLE_EVENT = "DomainAvailable";
-  String DOMAIN_COMPLETED_EVENT = "DomainCompleted";
-  String DOMAIN_PROCESSING_FAILED_EVENT = "DomainProcessingFailed";
-  String DOMAIN_PROCESSING_ABORTED_EVENT = "DomainProcessingAborted";
-  String DOMAIN_ROLL_COMPLETED_EVENT = "DomainRollCompleted";
-  String DOMAIN_ROLL_STARTING_EVENT = "DomainRollStarting";
-  String DOMAIN_VALIDATION_ERROR_EVENT = "DomainValidationError";
+  String DOMAIN_CHANGED_EVENT = "Changed";
+  String DOMAIN_CREATED_EVENT = "Created";
+  String DOMAIN_DELETED_EVENT = "Deleted";
+  String DOMAIN_AVAILABLE_EVENT = "Available";
+  String DOMAIN_COMPLETED_EVENT = "Completed";
+  String DOMAIN_FAILED_EVENT = "Failed";
+  String DOMAIN_ROLL_COMPLETED_EVENT = "RollCompleted";
+  String DOMAIN_ROLL_STARTING_EVENT = "RollStarting";
   String POD_CYCLE_STARTING_EVENT = "PodCycleStarting";
+  String DOMAIN_UNAVAILABLE_EVENT = "Unavailable";
+  String DOMAIN_INCOMPLETE_EVENT = "Incomplete";
+  String DOMAIN_FAILURE_RESOLVED_EVENT = "FailureResolved";
   String EVENT_NORMAL = "Normal";
   String EVENT_WARNING = "Warning";
   String WEBLOGIC_OPERATOR_COMPONENT = "weblogic.operator";
 
   String DOMAIN_AVAILABLE_PATTERN = "Domain %s became available";
-  String DOMAIN_CREATED_PATTERN = "Domain resource %s was created";
-  String DOMAIN_CHANGED_PATTERN = "Domain resource %s was changed";
+  String DOMAIN_CREATED_PATTERN = "Domain %s was created";
+  String DOMAIN_CHANGED_PATTERN = "Domain %s was changed";
   String DOMAIN_COMPLETED_PATTERN = "Domain %s is completely ready";
-  String DOMAIN_DELETED_PATTERN = "Domain resource %s was deleted";
-  String DOMAIN_PROCESSING_COMPLETED_PATTERN =
-      "Successfully completed processing domain resource %s";
-  String DOMAIN_PROCESSING_FAILED_PATTERN
-      = "Failed to complete processing domain resource %s due to: %s, the processing will be retried if needed";
-  String DOMAIN_PROCESSING_ABORTED_PATTERN
-      = "Aborting the processing of domain resource %s permanently due to: %s";
-  String DOMAIN_VALIDATION_ERROR_PATTERN
-      = "Validation error in domain resource %s: %s";
+  String DOMAIN_DELETED_PATTERN = "Domain %s was deleted";
+  String DOMAIN_FAILED_PATTERN = "Domain %s failed due to '%s': %s. %s";
+  String DOMAIN_UNAVAILABLE_PATTERN
+      = "Domain %s became unavailable because none of the servers is up although some of them are supposed to be up";
+  String DOMAIN_INCOMPLETE_PATTERN
+      = "Domain %s became incomplete because some of the servers is not up although they are supposed to be up";
```

Comment:
Domain %s is incomplete for one or more of the following reasons: failed is true, there are pending server shutdowns, or not all servers expected to be running are ready and at their target image, auxiliary images, restart version, and introspect version.

Reply:
done. thanks